lib/FrontendTool/FrontendTool.cpp

```diff
@@ -592,26 +594,33 @@ static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
   if (Invocation.getSILOptions().LinkMode == SILOptions::LinkAll)
     performSILLinking(SM.get(), true);
 
-  {
-    SharedTimer timer("SIL verification (pre-optimization)");
-    SM->verify();
+  if (!PrimarySourceFile) {
+    SM->linkAll();
+    SM->dump();
```

Comment:
I'm not sure what's being proposed, but `-disable-sil-perf-optzns` should be analogous to `-disable-llvm-optzns`. It should only disable optional SIL passes and should not affect functionality.